Objects/cellobject.c

```diff
@@ -74,6 +74,13 @@ PyCell_Set(PyObject *op, PyObject *value)
     }
 
     PyObject *old_value = PyCell_GET(op);
+
+    // Check that the reference can be created
+    if (!Py_REGIONCHANGEREFERENCE(op, old_value, value)) {
+        // Propagate the error from attempting to add the reference
+        return -1;
+    }
+
     PyCell_SET(op, Py_XNewRef(value));
```

Comment:
Could there be any issues here that the edge still exists when the region remove reference occurs?

Reply:
I don't think so. The invariant will only run when the bytecode is done executing. If a different thread has access to the Cell and gains access to the object between the `Py_REGIONCHANGEREFERENCE` and `PyCell_SET` we have a different problem on our hands.

I believe this should be safe, as long as region remove reference only decreases the LRC and doesn't try to verify anything